content/docs/09-registries-and-packs/4.5-create-pack.md

```diff
@@ -0,0 +1,843 @@
+
+<br /> <br />
+
+![Visual of generating an API key in Palette.](/tutorials/create-pack/registries-and-packs_create-pack_generate-api-key.png )
+
+<br />
+
+Here are the steps to add and export an API key:
+
+1. Fill in the required fields, such as the API key name and expiration date, and confirm to add the key. 
+
+2. Copy the key value to your clipboard, and switch back to the tutorials container environment. 
+
+3. Export the API key as an environment variable in the tutorials container bash session so the Terraform code can authenticate with Palette API.
+  <br />
+
+  ```bash
+  export SPECTROCLOUD_APIKEY=<Your-Spectro-Cloud-API-key>
+  ```
+<br />
+
+## Review Terraform Files
+Ensure that you have the following files in the current working directory.
+<br />
+
+```bash
+.
+├── profile.tf		# Resource
+├── cluster.tf		# Resource
+├── data.tf			  # Spectro Cloud data resources
+├── inputs.tf		  # Input varibales
+├── outputs.tf		# Output variables
+└── provider.tf		# Spectro Cloud Terraform provider 
+```
+
+Note that the Terraform code will deploy the resources to **AWS**. We recommend that you explore all Terraform files. Here is a high-level overview of each file:
+
+-  **profile.tf** - has the configuration for `spectrocloud_cluster_profile` resource. Review the core infrastructure layers that comprise the `spectrocloud_cluster_profile` resource. 
+
+
+-  **cluster.tf** - has the configuration for `spectrocloud_cluster_aws` resource. The cluster resource depends upon the `spectrocloud_cluster_profile` resource. 
+
+
+- **data.tf** - contains the configuration for the resources to retrieve data from Palette dynamically. For deploying the `spectrocloud_cluster_profile` resource, here are the pack details for each infrastructure layer. 
+
+  |**Pack Type**|**Registry**|**Pack Name**|**Tag**| **Version** |
+  |---|---|---|---|
+  |OS|Public Repo|`ubuntu-aws`|`LTS__20.4.x`| `20.04`|
+  |Kubernetes|Public Repo|`kubernetes`|`1.24.x`| `1.24.10` |
+  |Network|Public Repo|`cni-calico`|`3.25.x`|`3.25.0`|
+  |Storage|Public Repo|`csi-aws-ebs`|`1.16.x`|`1.16.0`|
+
+  Be aware that the Terraform code, when executed, will deploy the resources to the AWS cloud service provider. To deploy your resource to Azure or Google Cloud, use the specific layer details outlined in the **Cloud Service Provider Configurations** section [here](https://github.com/spectrocloud/tutorials/tree/main/terraform/pack-tf/README.md). 
+
+
+
+- **inputs.tf** - contains all the Terraform variables and the default values used in the tutorial. Before executing the terraform commands, verify and update the default value for all variables. In particular,  update the value for the `cluster_cloud_account_aws_name`, `ssh_key_name`, and `private_pack_registry` variables. You will find the `# ToDo` tags against the fields you want to update. Leave the remaining fields as default, assuming you will continue deploying to the AWS cloud. 
```

Comment:
I might be useful to have the user either create or modify a`terraform.tfvars`  file and show them to change the values. That way the chances of a user skipping this step reduces. Below is an example of what the `terraform.tfvars` file would contain.
```terraform
cluster_cloud_account_aws_name = "REPLACE ME"
aws_region_name = "REPLACE ME"
ssh_key_name = "REPLACE ME"
```

You can read more about tfvars files [here](https://developer.hashicorp.com/terraform/language/values/variables#variable-definitions-tfvars-files).

That way, you can remove the default value in `cluster_cloud_account_aws_name` `ssh_key_name` and `private_pack_registry`

It's a bit dangerous to use `default` values for these variables as the terraform run will break due to the values not being correct for the user. In those scenarios, `default` values should be avoided and instead, we can help the user by providing them a `terraform.tfvars` file for them to change.